amr-wind/incflo.cpp

```diff
@@ -107,7 +107,8 @@ void incflo::init_amr_wind_modules()
         eqn->initialize();
     }
 
-    m_sim.pde_manager().fillpatch_state_fields(m_time.current_time());
+    m_sim.pde_manager().fillpatch_state_fields(
+        m_time.current_time()); // may be unnecessary
```

Comment:
I added comments here, because I think that the changes in PDE.H make this call unnecessary. The default PostSolveOp as specified in PDEOps.H is a fillpatch of the pde field. This would imply that any custom PostSolveOp should at least contain the fillpatch step (and some also have clipping and calculation of other variables). Now that the PostSolveOp is included in the initialize() and post_regrid() steps for every PDE, there shouldn't be a need to loop through each PDE and fillpatch its variable, which is what fillpatch_state_fields does. The additional nuance of setting the state of the variable is not used in the code, it always uses the default of the New field state, aka the current value.

Reply:
Good point. Though of course it is always scary to remove a fillpatch... the consequences could be dire. But I think that under these conditions, you can be confident that it happens. We could comment that out with some version of your text included to indicate that we don't need the fillpatch because it happens in post_solve. Just in case this does come back to bite us. And then some time down the road when we are confident it didn't bite us, we can remove the comment. Thoughts?

Reply:
sounds good! I think we may not be able to comment it out because of CI, and also these instances are the only ones where this function is used, which would mean we should remove the fillpatch_state_fields function. But I think we need to comment somewhere to mention the fillpatch being included through another means. I can check the reg tests and the test case that led to this bug fix. I'll also confirm from the code that every pde has a post_solve step with fillpatch.

Reply:
turns out that the fillpatch_state_fields function is still necessary. Most of it is not, but the part where a density fillpatch is performed for constant density simulations needs to be there. In my opinion, that should be inside the icns PDE and not in the overall PDE system, because the density field is created when icns is created. Plus, much of this involves the m_constant_density boolean, which tends to be forgotten in input files for multiphase problems because it isn't required (no consequences for the simulation, but leads to unnecessary copying). fixing that in a nice way is a big enough scope for another PR.

I did confirm with fcompare that the constant density fillpatch is the only non-redundant part of fillpatch_state_fields.

Reply:
Ok lmk if you need help on that.